### PR TITLE
bug-1126191-related-tiles

### PR DESCRIPTION
**Pull request by tspurway**
Related tiles code complete.  

@oyiptong r? - i believe this is it for splice - give it a close going over

### CODE REVIEW

**Comment by tspurway** [comment]
Note, @oyiptong :  I haven't pushed the changes to migrate RedShift for schema changes in this commit.  I wanted to do that as a separate PR, if that works with you.

**Comment by oyiptong** [comment]
Can you put the migrations in this PR? It would make it easier to review

**Comment by tspurway** [comment]
The migrations are here for postgres.  i can't figure out why the travis build is failing...

**Comment by oyiptong** [comment]
There seem to be 2 test failures

**Comment by oyiptong** [comment]
Can you run fab test locally?

**Comment by tspurway** [comment]
yes.  it works fine locally

**Comment by oyiptong** [comment]
I'll take a look on monday

**Comment by oyiptong** [comment]
This is being tracked in https://bugzilla.mozilla.org/show_bug.cgi?id=1150540

File: splice/models.py [hunk 1]
```diff
@@ -34,12 +34,31 @@ class Tile(db.Model):
...
 
+    created_at = db.Column(db.DateTime(), nullable=False, default=datetime.utcnow)
+
+
+class Adgroup(db.Model):
+    __tablename__ = "adgroups"
+
+    id = db.Column(db.Integer(), autoincrement=True, primary_key=True, info={"identity": [1, 1]})
     locale = db.Column(db.String(14), nullable=False)
+    # country_code = db.Column(db.String(2), nullable=False)
```
Comment (oyiptong) [inline splice/models.py]: nit: please remove commented code

File: splice/queries.py [hunk 2]
```diff
@@ -23,21 +38,27 @@ def tile_exists(target_url, bg_color, title, type, image_uri, enhanced_image_uri
...
         .filter(Tile.image_uri == image_uri)
         .filter(Tile.enhanced_image_uri == enhanced_image_uri)
-        .filter(Tile.locale == locale)
+        .filter(Adgroup.locale == locale)
+        .join(Adgroup.tiles)
         .order_by(asc(Tile.id))
-        .first()
     )
+    # if country_code != 'STAR':
+    #     results = results.filter(Adgroup.country_code == country_code)
```
Comment (oyiptong) [inline splice/queries.py]: nit: please remove commented code

File: splice/queries.py [hunk 3]
```diff
@@ -263,33 +286,53 @@ def insert_tile(target_url, bg_color, title, type, image_uri, enhanced_image_uri
 
     try:
         conn.execute(
+            text(
+                "INSERT INTO adgroups ("
+                "locale, created_at"
+                # "country_code, locale, created_at"
```
Comment (oyiptong) [inline splice/queries.py]: nit: please remove commented code

File: splice/queries.py [hunk 4]
```diff
@@ -263,33 +286,53 @@ def insert_tile(target_url, bg_color, title, type, image_uri, enhanced_image_uri
...
+                ") "
+                "VALUES ("
+                ":locale, :created_at"
+                # " :country_code, :locale, :created_at"
```
Comment (oyiptong) [inline splice/queries.py]: nit: please remove commented code

**Comment by oyiptong** [comment]
Reading through the code, it looks like adgroups are completely immutable.

i.e. if one was to add a site to an adgroup, a totally new adgroup would be created. This is different from what we discussed. Is that expected?

**Comment by tspurway** [comment]
@oyiptong you are right about immutability.  I ran into a corner case where mutable adgroups caused issues in the decision to create or change a tile/adgroup.  I think we will need finer grained control (ie. a campaign management UI) to address selectively mutable adgroups.

**Comment by oyiptong** [comment]
r+

**Comment by oyiptong** [comment]
actually r-

discovered an issue where the tiles distributions are not being generated correctly.

Here are a couple of issues:
* the geo/local distributions are empty; they contain no tiles.
* please name the geo/local distribution files prefixed with ```ag```

## adgroup file naming

https://moz-tiles-local.s3.amazonaws.com/desktop/STAR/en-US.6781fcbb1a9873a7217bddbbd72643c61d5451cf.json

becomes

https://moz-tiles-local.s3.amazonaws.com/desktop/STAR/en-US.ag.6781fcbb1a9873a7217bddbbd72643c61d5451cf.json

**Comment by oyiptong** [comment]
r+

**Pull request closed without merging**